### ISSUE TITLE
Deleting experiments

### ISSUE THREAD

**Issue by davidbending** [Azure/MachineLearningNotebooks#196]
Is it possible to delete individual experiments within a workspace.  This would help keep things tidier and free up unneeded resources.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 47d45e80-f383-aeab-3c36-78db3481325a
* Version Independent ID: 49573b7e-1193-c0d0-b2a1-c38eaaed943c
* Content: [azureml.core.Experiment class - Azure Machine Learning Python](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.experiment(class)?view=azure-ml-py)
* Content Source: [AzureML/docs-ref-autogen/azureml-core/azureml.core.Experiment(class).yml](https://github.com/MicrosoftDocs/MachineLearning-Python-pr/blob/live/AzureML/docs-ref-autogen/azureml-core/azureml.core.Experiment(class).yml)
* Service: **machine-learning**
* GitHub Login: @j-martens
* Microsoft Alias: **jmartens**

**Comment by heatherbshapiro** [comment]
It is currently not possible to delete an experiment within a workspace but we are working on adding the functionality in the next several months.

**Comment by mitchellspryn** [comment]
Is there an ETA for this?

**Comment by amsword** [comment]
Any update on this issue?

**Comment by skasturi** [comment]
@deepak-muniappan

**Comment by alexkayal** [comment]
Seriously there should have been an update on this already. My workspace is cluttered with months' worth of experiments and there should be a simple way to clean it up.

**Comment by richardleeaus** [comment]
Hoping for this to come sooner rather than later to clear clutter. There has to be a better way other than starting a new workspace.

**Comment by alieus** [comment]
Is there any updates on this issue? This is definitely a must have for us. We are cluttered by months of experiment runs

**Comment by nathanclemente** [comment]
+1 hoping to have an update on this soon

**Comment by qawnaoya** [comment]
+1 hoping

**Comment by Ridooo** [comment]
+1 hoping

**Comment by SeanMirchi** [comment]
Well seems like they've added something in the latest version -> `azureml-core-1.0.69`
Now Experiment class has an `archive` function.
You can use it like this:

```
    from azureml.core.experiment import Experiment
    ws = Workspace.from_config(auth=authenticate())
    ex = Experiment(ws, 'experiment_name')
    ex.archive()
```

This disables the experiment and then it is not visible in the UI anymore.

**Comment by oussamachelly** [comment]
+1 `delete()` method in the SDK

**Comment by b93rn** [comment]
+1 `run.delete()`method in the SDK as well.

**Comment by jarandaf** [comment]
I see this is planned for version 2.0. What is the release date estimate?

**Comment by lostmygithubaccount** [comment]
a limited private preview is starting this month, public preview date is not confirmed - @jpe316 @danielsc

**Comment by mrevow** [comment]
looks like the official [documentation ](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.experiment.experiment?view=azure-ml-py#delete-workspace--experiment-id-)is ahead of the release. It already shows a delete method on the Experiment object

**Comment by jarandaf** [comment]
Latest package version (1.19.0) has indeed a `delete` method but it does not seem to allow deleting experiments with runs:

```python
ipdb> Experiment.delete(ws, exp.id)
*** msrestazure.azure_exceptions.CloudError: Azure Error: UserError
Message: Only empty Experiments can be deleted. This experiment contains run(s)
```

**Comment by lostmygithubaccount** [comment]
preview is usable with limited support and features (but basic and distributed training works) - you can find the docs at https://aka.ms/azuremlv2 and the source repo at https://github.com/Azure/azureml-v2-preview

**Comment by swinner95** [comment]
While deleting experiments is currently not supported from the AML studio UI, the functionality to delete runs from the AML Studio UI is currently under private preview and should be available for public preview ETA end of of January.

**Comment by ncolossi** [comment]
I have been able to see the AML studio UI delete functionality on my workspace. However my customer does not see it yet. Do we need to whitelist their workspace for them to see the Delete button? Can you confirm is public preview?

**Comment by lostmygithubaccount** [comment]
often, Microsoft employees will see these types of UI previews shortly before being rolled out publicly - @swinner95 to confirm ETAs/if you can onboard a customer sooner

**Comment by swinner95** [comment]
the run delete is still under private preview where users can delete the run metadata and metrics.  For this functionality to be enabled for your subscription,  users will need to request access and we will enable it for your subscription(s). (Signup:  https://forms.office.com/r/57kj1BPx1t ) We are working on improving this functionality before rolling this functionality out more broadly.

**Comment by slsu0424** [comment]
Hello - I seem to be running into a similar issue as @jarandaf.  Following the available syntax here - https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.experiment.experiment?view=azure-ml-py#delete-workspace--experiment-id- 

I ran the code in Azure ML with the following error:

```
azureuser@ssu1:~/cloudfiles/code/Users/sansu$ python sample_code.py 
Traceback (most recent call last):
  File "sample_code.py", line 3, in <module>
    delete(ssuaml, mlflow-diabetes-experiment-logreg)
NameError: name 'delete' is not defined
```

**Comment by swinner95** [comment]
Can you try calling  Experiment.delete(ssuaml,  mlflow-diabetes-experiment-logreg) ?

**Comment by slsu0424** [comment]
@swinner95  - here is the error I receive:

python sample_code.py 
azureuser@ssu1:~/cloudfiles/code/Users/sansu$ 
azureuser@ssu1:~/cloudfiles/code/Users/sansu$ python sample_code.py 
Traceback (most recent call last):
  File "sample_code.py", line 4, in <module>
    Experiment.delete(ssuaml, mlflow-diabetes-experiment-logreg)
NameError: name 'Experiment' is not defined
azureuser@ssu1:~/cloudfiles/code/Users/sansu$

**Comment by edwardnguyen1705** [comment]
Has thi issue been solved yet? 
I have encountered something like this:
Here is what I have been trying to do:

```
# get a list of all Experiments objects contained in Workspace
experiments_lst = Experiment.list(ws)
for experiment in experiments_lst:
    print(experiment.name)
    Experiment.delete(ws, experiment.id)
```

And the response looks like this:

```
    raise CloudError(self._response)
msrestazure.azure_exceptions.CloudError: Azure Error: UserError
Message: Only empty Experiments can be deleted. This experiment contains run(s)
```

**Comment by AnaValeriaS** [comment]
Any update on this issue?

**Comment by luigiw** [comment]
Hello, ```Experiment.delete()``` can only delete empty experiments, i.e. experiments without any runs. So if your experiment is not empty, you'll need to delete all the runs in it first before deleting the experiment.

**Comment by coleman-wma** [comment]
Running into the same 'experiment can't be deleted because it's got runs in it' message. It's not apparent how to delete runs - no link that I can see in the UI, the run object doesn't have a delete option - can someone tell me how runs can be deleted?

**Comment by jarandaf** [comment]
AFAIK runs can be deleted from the UI in private preview. I am not sure whether this feature has been already officially released.

**Comment by coleman-wma** [comment]
Thanks @jarandaf - I've requested that private preview as I don't have any run delete option in the UI.

**Comment by ZekunZh** [comment]
Hello, I am having the same issue for deleting the experiments and the runs in one experiment.

# What I have tried: 3 different ways to delete the run

1. For each run in the experiment, I ran `run.clean()`. Then I ran `Experiment.delete(workspace, experiment_id)`.
    - Result: **nothing have changed** in Azure blob. I can find all the outputs in the blob `AzmlStorageAccount/azureml/ExperimentRun/dcid.{my_run_id}`

1. In the UI of azureml, I deleted all the runs.
    - Result: **nothing have changed** in Azure blob. I can find all the outputs in the blob `AzmlStorageAccount/azureml/ExperimentRun/dcid.{my_run_id}`

1. In the blob storage of azureml, I deleted all the blobs that are concerned with the run, mainly those:
    1. `AzmlStorageAccount/azureml/ExperimentRun/dcid.{my_run_id}`
    2. `AzmlStorageAccount/azureml/ArtifactZip/ExperimentRun/dcid.{my_run_id}.zip`
    3. `AzmlStorageAccount/azureml/ComputeRecord/dcid.{my_run_id}`
    
    - Result: nothing have changed in UI of azureml. We still have access to `Outputs+logs` and `Code`
    

In summary, currently the UI interface is discorrelated with the Azure blob storage. And the Python APIs for experiments & runs have no effect neither on UI interface nor on Azure blob storage.

I will be really grateful if anyone can give some ideas. Thanks !

**Comment by yokosyun** [comment]
> +1 `run.delete()`method in the SDK as well.

it does not exist run.delete() nor run.archive()  with azureml-sdk=1.55.0
